src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -111,6 +111,10 @@ public function __clone()
     public function boot()
     {
         if (true === $this->booted) {
+            if ($this->debug) {
+                $this->startTime = microtime(true);
+            }
```

Comment:
I would remove the initialization done currently in `__clone()` as this is now redundant with this one.

Also, to be consistent, what about removing the same code in `__construct()` and always execute the initialization when the `boot()` method is called?

Reply:
As a side note, there is a side effect of this change (but this was already the case as of 3.4). Before 3.4, calling `boot()` a second time was a no-op. As of 3.4, calling `boot()` a second time re-initialize the Kernel.

Reply:
@fabpot Fixed.